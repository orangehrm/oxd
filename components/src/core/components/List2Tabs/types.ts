export type Tab = {
  id: string;
  label: string;
  icon: string;
};
