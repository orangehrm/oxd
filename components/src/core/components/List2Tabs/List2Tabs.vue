<template>
  <div class="oxd-list-tab">
    <div class="oxd-list-tab-content">
      <oxd-text
        v-for="tab in computedTabs"
        :key="tab.id"
        :id="tab.id"
        :class="{
          'oxd-list-tab-item': true,
          '--selected': modelValue === tab.id,
        }"
        tag="span"
        @click="$emit('update:modelValue', tab.id)"
      >
        <oxd-icon :name="tab.icon" size="large"></oxd-icon>
        {{ tab.label }}
      </oxd-text>
      <div class="oxd-list-tab-buttons">
        <slot></slot>
      </div>
    </div>
    <oxd-divider class="oxd-list-tab-divider" />
  </div>
</template>

<script lang="ts">
import {nanoid} from 'nanoid';
import {Tab} from './types';
import {computed, defineComponent, PropType} from 'vue';
import Icon from '@orangehrm/oxd/core/components/Icon/Icon.vue';
import Text from '@orangehrm/oxd/core/components/Text/Text.vue';
import Divider from '@orangehrm/oxd/core/components/Divider/Divider.vue';

export default defineComponent({
  name: 'oxd-list-tabs',

  components: {
    'oxd-icon': Icon,
    'oxd-text': Text,
    'oxd-divider': Divider,
  },

  emits: ['update:modelValue'],

  props: {
    tabs: {
      type: Array as PropType<Tab[]>,
      default: () => [],
    },
    modelValue: {
      type: String,
      default: null,
    },
  },

  setup(props) {
    const computedTabs = computed(() =>
      props.tabs.map(tab => ({
        id: nanoid(),
        ...tab,
      })),
    );

    return {
      computedTabs,
    };
  },
});
</script>

<style lang="scss" src="./list2-tabs.scss" scoped></style>
