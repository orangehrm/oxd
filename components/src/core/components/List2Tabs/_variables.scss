@import '../List2/variables';
