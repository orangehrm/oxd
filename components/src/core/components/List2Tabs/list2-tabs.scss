@import './variables';
@import '../../../styles';

.oxd-list-tab {
  padding-left: $oxd-list-padding;
  padding-right: $oxd-list-padding;
  background-color: $oxd-white-color;
  border-top-left-radius: $oxd-border-radius;
  border-top-right-radius: $oxd-border-radius;

  &-content {
    display: flex;
    align-items: center;
    padding: calc(#{$oxd-list-padding}* 0.5) 0;
  }

  &-item {
    gap: 0.2rem;
    cursor: pointer;
    position: relative;
    font-weight: 700;
    margin-right: 5px;
    padding: 8px 15px;
    font-size: 0.875rem;
    align-items: center;
    display: inline-flex;

    &:hover,
    &.--selected {
      border-radius: 2.5rem;
      background-color: #f1f2f5;
    }

    &.--selected::after {
      content: '';
      left: 0;
      width: 30%;
      height: 3px;
      bottom: -12px;
      position: absolute;
      background-color: #ff6000;
    }
  }

  &-buttons {
    margin-left: auto;
  }

  &-divider {
    padding-bottom: 10px;
  }
}
