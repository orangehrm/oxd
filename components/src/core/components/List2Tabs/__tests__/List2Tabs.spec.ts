import {mount} from '@vue/test-utils';
import List2Tabs from '@orangehrm/oxd/core/components/List2Tabs/List2Tabs.vue';

describe('List2 > ListTabs.vue', () => {
  it('renders List Tabs', () => {
    const wrapper = mount(List2Tabs, {
      props: {
        tabs: [
          {id: 'goal', label: 'Goal', icon: 'oxd-chat'},
          {id: 'okrs', label: 'OKRs', icon: 'oxd-search'},
        ],
        modelValue: 'okrs',
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
