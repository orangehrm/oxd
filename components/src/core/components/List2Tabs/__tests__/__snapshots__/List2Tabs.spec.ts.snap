// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List2 > ListTabs.vue renders List Tabs 1`] = `
<div class="oxd-list-tab">
  <div class="oxd-list-tab-content"><span class="oxd-text oxd-text--span oxd-list-tab-item"><i class="oxd-svg-icon oxd-svg-icon--large"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 60 60" style="enable-background:new 0 0 60 60" xml:space="preserve"><path style="fill:currentColor" d="M30 1.5c-16.542 0-30 12.112-30 27 0 5.204 1.646 10.245 4.768 14.604-.591 6.537-2.175 11.39-4.475 13.689a1 1 0 0 0 .846 1.697c.405-.057 9.813-1.411 16.618-5.339C21.621 54.71 25.737 55.5 30 55.5c16.542 0 30-12.112 30-27s-13.458-27-30-27zm-14 31c-2.206 0-4-1.794-4-4s1.794-4 4-4 4 1.794 4 4-1.794 4-4 4zm14 0c-2.206 0-4-1.794-4-4s1.794-4 4-4 4 1.794 4 4-1.794 4-4 4zm14 0c-2.206 0-4-1.794-4-4s1.794-4 4-4 4 1.794 4 4-1.794 4-4 4z"></path></svg></i> Goal</span><span class="oxd-text oxd-text--span oxd-list-tab-item --selected"><i class="oxd-svg-icon oxd-svg-icon--large"><svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path style="fill:currentColor" d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"></path></svg></i> OKRs</span>
    <div class="oxd-list-tab-buttons"></div>
  </div>
  <div class="oxd-list-tab-divider">
    <hr class="oxd-divider-horizontal" role="separator" aria-orientation="horizontal">
    <!--v-if-->
  </div>
</div>
`;
